src/tests/Microsoft.PowerFx.Connectors.Tests/BasicRestTests.cs

```diff
@@ -137,5 +142,165 @@ public async Task PetStore_MultiServer()
+
+            // https://math.tools/api/numbers/
+            // Note that these APIs are rate limited and HTTP error 429 is possible
+            var swaggerUrl = "https://api.apis.guru/v2/specs/math.tools/1.5/openapi.json";
+
+            OpenApiDocument doc = await ReadSwaggerFromUrl(swaggerUrl).ConfigureAwait(false);
+
+            // Here we specify the BaseAddress
+            using var client = new HttpClient() { BaseAddress = new Uri("https://api.math.tools") };
+
+            // Set IgnoreUnknownExtensions to true as this swagger uses some extensions we don't honnor like x-apisguru-categories, x-origin, x-providerName
+            var funcs = config.AddService("Math", doc, client, new ConnectorSettings() { IgnoreUnknownExtensions = true });
+
+            var engine = new RecalcEngine(config);
+            var expr = "Math.numbersbasebinary(632506623)";
+            var check = engine.Check(expr);
+            var ok = check.IsSuccess;
+
+            Assert.True(ok, string.Join(", ", check.Errors.Select(er => er.Message)));
+
+            FormulaValue result = engine.Eval(expr);
+
+            if (result is ErrorValue ev)
+            {
+                Assert.True(false, string.Join(", ", ev.Errors.Select(er => er.Message)));
+            }
+
+            // To read the complete result
+            var sb = new StringBuilder();
+            result.ToExpression(sb, new FormulaValueSerializerSettings() { UseCompactRepresentation = true });
+            string resultExpr = sb.ToString();
+
+            // Create an object from the result and access it
+            dynamic dResult = result.ToObject();
+            string bin = dResult.contents.answer;
+
+            Assert.Equal("100101101100110100100011111111", bin);
+
+            // Try same number but in base 5 (why not??)
+            var expr2 = @"Math.numbersbasebinary(2243410202443, {from: 35})";
+            result = engine.Eval(expr2);
+
+            if (result is ErrorValue ev2)
+            {
+                Assert.True(false, string.Join(", ", ev2.Errors.Select(er => er.Message)));
+            }
+
+            dynamic dResult2 = result.ToObject();
+            string bin2 = dResult.contents.answer;
+
+            Assert.Equal("100101101100110100100011111111", bin2);
+        }        
+
+        [Fact]
+        public async Task RealTest3()
+        {
+            var config = new PowerFxConfig();
+
+            // https://date.nager.at/       
+            var swaggerUrl = "https://date.nager.at/swagger/v3/swagger.json";
+
+            OpenApiDocument doc = await ReadSwaggerFromUrl(swaggerUrl).ConfigureAwait(false);
+            using var client = new HttpClient() { BaseAddress = new Uri("https://date.nager.at") };            
+            var funcs = config.AddService("Holiday", doc, client);
+
+            var engine = new RecalcEngine(config);            
+            var expr = @"Holiday.PublicHolidayPublicHolidaysV3(2023, ""US"")";
```

Comment:
>Holiday.PublicHolidayPublicHolidaysV3(2023, ""US"") [](http://example.com/codeflow?start=25&length=51)

@LucGenetier - 
Trying to use return value gets a runtime exception.  Unlike above, we do have a type, so not sure where it's wrong. 

Try this expression:
First(Holiday.PublicHolidayPublicHolidaysV3(2023, ""US"")).localName

passes check, but fails at runtime.